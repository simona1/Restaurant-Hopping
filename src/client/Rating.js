import React from 'react';

export default class Rating extends React.Component<Props> {
  render() {
    const {rating, id} = this.props;
    const stars = [];
    for (let i = 0; i < rating; ++i) {
      stars.push(
        <i key={`${id}-${i}`} className="material-icons md-icon dp18">
          grade
        </i>,
      );
    }
    return (
      <span
        className="item-rating"
        style={{width: Math.floor(24 * rating) + 'px'}}>
        {stars}
      </span>
    );
  }
}
