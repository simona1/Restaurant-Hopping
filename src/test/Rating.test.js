import Rating from '../client/Rating';
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

const testProps = [
  {
    id: 'ChIJZ5XfV10GhYARDh2gyITiZGw',
    rating: 4.5,
  },
];

describe('Rating', () => {
  let component;
  beforeEach(() => {
    component = shallow(<Rating id={testProps.id} rating={testProps.rating} />);
  });
  it('should render a <Rating />', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
  it('should contain a span element', () => {
    expect(component.contains('span'));
  });
  it('has length', () => {
    expect(component.find('span').length).toEqual(1);
  });
  it('should contain nested elements', () => {
    expect(component.contains('i'));
  });
});
