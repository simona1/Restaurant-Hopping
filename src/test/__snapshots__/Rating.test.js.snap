// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should render a <Rating /> 1`] = `
<span
  className="item-rating"
  style={
    Object {
      "width": "NaNpx",
    }
  }
/>
`;
